src/test/java/org/apache/sysds/test/functions/federated/primitives/FederatedCtableTest.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.functions.federated.primitives;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.sysds.common.Types;
+import org.apache.sysds.runtime.meta.MatrixCharacteristics;
+import org.apache.sysds.runtime.util.HDFSTool;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(value = Parameterized.class)
+@net.jcip.annotations.NotThreadSafe
+public class FederatedCtableTest extends AutomatedTestBase {
+	private final static String TEST_DIR = "functions/federated/";
+	private final static String TEST_NAME1 = "FederatedCtableTest";
+	private final static String TEST_NAME2 = "FederatedCtableFedOutput";
+	private final static String TEST_CLASS_DIR = TEST_DIR + FederatedCtableTest.class.getSimpleName() + "/";
+
+	private final static int blocksize = 1024;
+	@Parameterized.Parameter()
+	public int rows;
+	@Parameterized.Parameter(1)
+	public int cols;
+	@Parameterized.Parameter(2)
+	public int maxVal1;
+	@Parameterized.Parameter(3)
+	public int maxVal2;
+	@Override
+	public void setUp() {
+		TestUtils.clearAssertionInformation();
+		addTestConfiguration(TEST_NAME1, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME1, new String[] {"F"}));
+		addTestConfiguration(TEST_NAME2, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME2, new String[] {"F"}));
+	}
+
+	@Parameterized.Parameters
+	public static Collection<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+			{12, 4, 4, 7},
+//			{100, 14, 4, 7}, {1000, 14, 4, 7}
+		});
+	}
+
+	@Test
+	public void federatedCtableSinglenode() {
+		runCtable(Types.ExecMode.SINGLE_NODE, false);
+	}
+
+	@Test
+	public void federatedCtableFedOutputSinglenode() {
+		runCtable(Types.ExecMode.SINGLE_NODE, true);
+	}
```

Comment:
I think we could use more test cases. 
How about a test case where Y is federated and X is not? 
What about a third input? (especially something that triggers the reversedWeights-case in the processInstruction method)
It is also relevant with some invalid input to see if it actually throws an exception or if it computes meaningless output (for instance invalid dimensionality of the input). 
I think it is also very relevant to test with different federated range overlaps, since the slicing is an important component of the CTable instruction.

Reply:
I added more tests now and fixed reversed weights.